flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/sources/TableSourceUtil.scala

```diff
@@ -266,17 +267,22 @@ object TableSourceUtil {
         // push an empty values node with the physical schema on the relbuilder
         relBuilder.push(createSchemaRelNode(resolvedFields))
         // get extraction expression
-        resolvedFields.map(f => PlannerResolvedFieldReference(f._1, f._3))
+        resolvedFields.map(f => new ResolvedFieldReference(f._1, f._3, f._2))
       } else {
-        new Array[PlannerResolvedFieldReference](0)
+        new Array[ResolvedFieldReference](0)
       }
 
-      val expression = tsExtractor
-        .getExpression(fieldAccesses.map(_.asInstanceOf[ResolvedFieldReference]))
+      val expression = tsExtractor.getExpression(fieldAccesses)
       // add cast to requested type and convert expression to RexNode
-      // TODO we cast to planner expressions as a temporary solution to keep the old interfaces
```

Comment:
readd  this TODO?

Reply:
OK, I will add `TODO we convert to planner expressions as a temporary solution`